src/ansys/grantami/jobqueue/_models.py

```diff
@@ -138,61 +138,101 @@ def to_dict(self) -> Dict[str, Union[int, None]]:
         }
 
 
-class _JobFile:
+class JobFile:
     """
-    Represents a file associated with a job request.
+    Represents a file associated with a JobRequest.
 
     Parameters
     ----------
-    file_type : _FileType
-        Type of file being represented.
+    path : str or pathlib.Path
+        Path to the local file to use in a JobRequest. Can be absolute or relative.
+    virtual_path : str or pathlib.Path
+        Virtual path to use to refer to the file in the JobRequest. Must be relative.
     """
 
-    def __init__(self, file_type: _FileType):
-        self.file_type = file_type
-        self._path: Union[pathlib.Path, None] = None
-        self._id: Optional[str] = None
+    def __init__(self, path: Union[str, pathlib.Path], virtual_path: Union[str, pathlib.Path]):
+        self._path: pathlib.Path = pathlib.Path(path)
+        virtual_path = pathlib.Path(virtual_path)
```

Comment:
Could we require a virtual path to always be a pathlib.Path? It seems like this would be a minor inconvenience for the user, and it would reduce the surface area we need to worry about.

Reply:
I've left the option to use strings as in most cases a name is sufficient. This enables for example `JobFile(combined_data_file_path, "data.xlsx")`.
If we manage to catch simple cases so that they don't require the use of JobFile aymore then I agree that only Path would be better